File: Programming EF Code First/chapter5/Chapter 5 VB/Fluent/DataAccess/PersonConfiguration.vb
Imports System.Data.Entity.ModelConfiguration
Imports Model
Imports System.ComponentModel.DataAnnotations

Namespace DataAccess
  Public Class PersonConfiguration
	  Inherits EntityTypeConfiguration(Of Person)
	Public Sub New()
      Me.Property(Function(p) p.SocialSecurityNumber).IsConcurrencyToken()

      Me.Property(Function(p) p.Address.StreetAddress).HasColumnName("StreetAddress")

      Me.Property(Function(p) p.Address.City).HasColumnName("City")

      Me.Property(Function(p) p.Address.State).HasColumnName("State")

      Me.Property(Function(p) p.Address.ZipCode).HasColumnName("ZipCode")

      Me.ToTable("People")
	End Sub
  End Class
End Namespace

File: Programming EF Code First/chapter5/Chapter5 CSharp/BAGA_FluentAPI/DataAccess/PersonConfiguration.cs
using System.Data.Entity.ModelConfiguration;
using Model;
using System.ComponentModel.DataAnnotations;

namespace DataAccess
{
  public class PersonConfiguration : EntityTypeConfiguration<Person>
  {
    public PersonConfiguration()
    {
      Property(p => p.SocialSecurityNumber)
        .IsConcurrencyToken();

      Property(p => p.Address.StreetAddress)
        .HasColumnName("StreetAddress");

      Property(p => p.Address.City)
        .HasColumnName("City");

      Property(p => p.Address.State)
        .HasColumnName("State");

      Property(p => p.Address.ZipCode)
        .HasColumnName("ZipCode");

      ToTable("People");
    }
  }
}